web_client/components/body/analysis/element/feature-selection.jsx

```diff
@@ -35,200 +43,194 @@ class FeatureSelection extends React.Component {
+          extendedText: ''
+        });
+      }
     };
 
     return (
-      <div>
+      <div
+        className='feature-select'
+        style={{
+          alignItems: 'center',
+          display: 'flex',
+          paddingBottom: 5,
+          paddingTop: 5
+        }}
+      >
         <div
-          className='col-md-6'
+          className='input-group'
           style={{
-            paddingRight: 0
-            // verticalAlign: 'top',
-            // marginRight: 2
+            width: '45%',
+            marginRight: '0.5%'
           }}
         >
-          <label>Search for features</label>
-          <br />
-          <div
-            className='col-md-10'
-            style={{
-              marginRight: 0,
-              paddingRight: 0
-            }}
-          >
-            <input
-              className='form-control'
-              list='types'
-              onChange={(e) => {
-                let { target: { value: currentText } } = e;
-                onStateChange({ currentText });
-                onAction(action, currentText);
-              }}
-
-              onKeyPress={(e) => {
-                if (e.key === 'Enter') {
-                  e.preventDefault();
-                  handleNewEntries();
-                }
-              }}
-
-              placeholder='Enter Feature'
+          <span className='input-group-btn'>
+            <button
+              className='btn btn-default'
+              type='button'
               style={{
-                // width: 200,
-                // display: 'inline',
-                // verticalAlign: 'top'
+                height: (singleMode ? 33 : 100),
+                marginBottom: 0,
+                marginTop: 0,
+                minWidth: 70,
+                transition: 'height 0.1s ease'
               }}
-              type='search'
-              value={currentText}
-            />
-            <datalist id='types'>
-              {candidates.map(({id, description}) => (
-                <option key={id} value={id}>{description}</option>
-              ))}
-            </datalist>
-          </div>
-          <div
-            className='col-md-2'
-            style={{
-              marginRight: 0,
-              padding: 0
-            }}
-          >
-            <button
-              className='btn btn-primary'
               onClick={(e) => {
                 e.preventDefault();
-                handleNewEntries();
-              }}
-              style={{
-                margin: 0
-                // marginRight: 1,
-                // paddingTop: 8,
-                // paddingLeft: 7,
-                // height: 33,
-                // width: 35,
-                // verticalAlign: 'top'
+                onStateChange({
+                  singleMode: !singleMode,
+                  transitioning: true
+                });
               }}
             >
-              <i className='icon-plus' />
+              {
+                transitioning
```

Comment:
This looks a lot like it could simplified to `transitioning === singleMode ? 'Multi' : 'Search"` but that may not be any less confusing.